src/main/java/io/gravitee/el/spel/context/SpelTemplateContext.java

```diff
@@ -49,4 +77,47 @@ public Object lookupVariable(String name) {
     public EvaluationContext getContext() {
         return context;
     }
+
+    public Single<EvaluationContext> evaluationContext(CachedExpression expression) {
+        if (deferredVariables != null) {
+            final List<Completable> collect = deferredVariables
+                .entrySet()
+                .stream()
+                .filter(deferredEntry -> requiresDeferredVariable(expression, deferredEntry))
+                .map(e -> resolveDeferred(e.getKey(), e.getValue()))
+                .collect(Collectors.toList());
+
+            return Completable.merge(collect).andThen(Single.just(context));
+        }
+
+        return Single.just(context);
+    }
+
+    private boolean requiresDeferredVariable(CachedExpression expression, Map.Entry<String, Object> deferredEntry) {
+        // Ex: expression: 'request.contentJson.test', variable 'request.contentJson'
+        return expression
+            .getVariables()
+            .stream()
+            .anyMatch(variable -> variable.equals(deferredEntry.getKey()) || variable.startsWith(deferredEntry.getKey() + "."));
```

Comment:
should'nt you just test variable.startsWith(deferredEntry.getKey()) ?

Reply:
Not exactly. The idea is to cover the 2 following cases

Lets say you have the following expression  `{#request.content}` (just the body content as string)
and the template context has a deferred variable which is `templateContext.setDeferredVariable("request.content", aCompletable);`

Here we are covering the case where `variable.equals(deferredEntry.getKey())`.

Now, lets say you have the following expression` {#request.jsonContent.foo.bar}`
and the template context has a deferred variable which is `templateContext.setDeferredVariable("request.jsonContent", aCompletable);`

Here we are covering the second case where `variable.startsWith(deferredEntry.getKey() + ".")` because `request.jsonContent.foo.bar` starts with `request.jsonContent.` (with a dot '.')

We cannot just have `startsWith` because it would lead to invalid resolution such as `request.jsonContentThatDoesExist` which starts with `request.jsonContent` but is not a real access to this variable.

Does it make sense to you ?

Reply:
yes perfectly